### ISSUE TITLE
Create s3 objects tags in aws govcloud partition

### ISSUE THREAD

**Issue by enxebre** [terraform-providers/terraform-provider-aws#2664]
### Terraform Version
Terraform v0.10.7

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_s3_bucket_object

### Expected Behavior
Should be able to add s3 objects tags in aws govcloud partition

### Actual Behavior
Got error message "This region does not allow for tags on S3 objects"

### Terraform Configuration Files
```hcl
resource "aws_s3_bucket" "meh" {
  bucket = "meh"
  acl = "private"
  tags = "${merge(map(
      "Name", "meh"
    ), var.extrag_tags)}"

  lifecycle {
    ignore_changes = ["*"]
  }
}
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Running on AWS GovCloud

### References
This is actually supported by GovCloud https://aws.amazon.com/about-aws/whats-new/2017/08/amazon-s3-storage-management-features/ 
This is being restricted in the code https://github.com/terraform-providers/terraform-provider-aws/blob/48fdbc0997746c9c226d36bf358d8884b176c0da/aws/resource_aws_s3_bucket_object.go#L202-L204

**Comment by apparentlymart** [comment]
Hi @enxebre! Thanks for reporting this.

I assume that at some earlier time there _were_ restrictions on using S3 tags in the govcloud partition, which motivated this extra check, but this restriction was subsequently lifted and so the Terraform provider now needs to be updated to respect that.

I see your corresponding pull request #2665 which addresses this. The Terraform team at HashiCorp is about to start our break for the holidays, but hopefully we can take a closer look at your PR in the new year. Thanks for working on that!

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
